src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -112,16 +112,15 @@ private static Long openThemeImpl(String widget, int dpi) {
        } else {
            theme = openTheme(widget, dpi);
        }
+       if (theme == null || theme == 0) {
+           theme = openTheme(widget, defaultDPI);
+       }
```

Comment:
`theme` can't be `null` because `openTheme` returns `long`. Perhaps, the declaration should be changed to
```java
long theme;
```

This is still incorrect. If `i > 0`, there's a prerequisite to calling `openTheme`. Likely, you need another helper method.

Reply:
I didn't get the need for helper method?

Reply:
Ok, I got it. And I have updated. But if we want to change `theme` to `long`, then I guess we have to do it every calling function I guess. So is it better to leave it as `Long` ?

Reply:
I still think we should change `Long` to `long`. Yes, you'll need to update it in the two methods. In `getThemeImpl`, it will be boxed into `Long` automatically before being put into the map.

Reply:
When the primitive type `long` is used, there's no confusion whether the value can be `null` or not.